group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -929,12 +929,22 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         ConsumerGroup group = getOrMaybeCreateConsumerGroup(groupId, false);
         ConsumerGroupMember member = group.getOrMaybeCreateMember(memberId, false);
 
-        log.info("[GroupId " + groupId + "] Member " + memberId + " left the consumer group.");
-
-        List<Record> records = consumerGroupFenceMember(group, member);
+        List<Record> records = new ArrayList<>();
+        // The departing member is a static one. We don't need to fence this member because it is
+        // expected to come back within session timeout
+        if (member.instanceId() != null) {
+            log.info("Member {} with instance id {} is a static member and will not be fenced from the group", memberId, member.instanceId());
+        } else {
+            log.info("[GroupId {}] Member {} left the consumer group.", groupId, memberId);
```

Comment:
I have taken the liberty and updated the log line to use an argument based loggers instead of the string concatenation based pattern that existed before.